### PR TITLE
IModList updates

### PR DESCRIPTION
**Pull request by Holt59**
See `uibase` PR: https://github.com/ModOrganizer2/modorganizer-uibase/pull/92

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -5133,9 +5133,7 @@ void MainWindow::on_actionSettings_triggered()
       ui->dataTabShowFromArchives->setCheckState(Qt::Checked);
       ui->dataTabShowFromArchives->setEnabled(true);
     }
-    m_OrganizerCore.refreshModList();
-    m_OrganizerCore.refreshDirectoryStructure();
-    m_OrganizerCore.refreshLists();
+    m_OrganizerCore.refresh();
   }
```
Comment (Al12rs) [inline src/mainwindow.cpp:5137]: Uh I guess we got some optimization here, nice one. Hopefully it wasn't needed for some convoluted reason.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -6551,7 +6549,7 @@ void MainWindow::sendSelectedModsToSeparator_clicked()
 
       int newPriority = INT_MAX;
       bool foundSection = false;
-      for (auto mod : m_OrganizerCore.modsSortedByProfilePriority()) {
+      for (auto mod : m_OrganizerCore.modsSortedByProfilePriority(m_OrganizerCore.currentProfile())) {
```
Comment (Al12rs) [inline src/mainwindow.cpp:6552]: Doesn't calling without argument already use the currentProfile?

Comment (Al12rs) [reply src/mainwindow.cpp:6552]: Ah no you are doing that only for the modlist. Wouldn't it be easier to make it work like that for organizerCore as well?

Comment (Holt59) [reply src/mainwindow.cpp:6552]: I thought about it but it was adding another default parameter and an extra check in `modsSortedByProfilePriority` which kind of feel redundant. Plus you always have access to `m_OrganizerCore.currentProfile()` if you are calling `m_OrganizerCore.modsSortedByProfilePriority()`.

Comment (Al12rs) [reply src/mainwindow.cpp:6552]: I guess so. Does it work properly for getting the order of other profiles? Like I'm not sure if the non active profile is loaded with all the needed stuff

File: src/mainwindow.cpp (hunk 1 above)
Comment (Holt59) [reply src/mainwindow.cpp:5137]: The call to `refreshDirectoryStructure` was likely doing nothing (unless the directory update from `refreshModList` had already finished... ) due to the `m_DirectoryUpdate` "protection". Even with a very small setup, I doubt the update would have finished when reaching this line.

The `refreshLists` was doing something, but with the outdated directory structure, and this was override by the call to `refreshLists` in the slot at the end of the directory structure refresh. To me this looks like a left-over from a previous implementation where `refreshModList()` would  not refresh the directory structure or to a mistake due to the name `refreshModList()`.

File: src/mainwindow.cpp (hunk 2 above)
Comment (Holt59) [reply src/mainwindow.cpp:6552]: I did some quick testing with a Python module but in theory it should work properly:

- The priority stuff (`getPriorityMinimum`, `numRegularMods`, `modIndexByPriority`) depends on the `m_ModIndexByPriority` /  `m_NumRegularMods` which are set in `Profile::updateIndices()`.
- `Profile::updateIndices()` is called in `Profile::refreshModStatus()`.
- `Profile::refreshModStatus()` is called in all `Profile` constructors.

Comment (Al12rs) [reply src/mainwindow.cpp:6552]: That is nice hear, this could allow some cool profile sync stuff for plugins

**Pull request merged**